samza-core/src/main/java/org/apache/samza/runtime/LocalContainerRunner.java

```diff
@@ -71,14 +72,29 @@ public void run(StreamApplication streamApp) {
       Object taskFactory = TaskFactoryUtil.createTaskFactory(config, streamApp, this);
 
       SamzaContainer container = SamzaContainer$.MODULE$.apply(
-          containerModel.getProcessorId(),
           containerModel,
           config,
           jobModel.maxChangeLogStreamPartitions,
-          SamzaContainer.getLocalityManager(containerId, config),
           jmxServer,
           Util.<String, MetricsReporter>javaMapAsScalaMap(new HashMap<>()),
           taskFactory);
+      container.setContainerListener(
+          new SamzaContainerListener() {
+            @Override
+            public void onContainerStart() {
+              log.info("Container Started");
+            }
+
+            @Override
+            public void onContainerStop(boolean invokedExternally) {
+              log.info("Container Stopped");
+            }
+
+            @Override
+            public void onContainerFailed(Throwable t) {
+              log.info("Container Failed");
```

Comment:
This needs to be handled for the yarn case. Previously container.run() will throw exception when there is a failure. Now seems the logic is changed and the exception is populated in the callback. So it seems we need to store the exception into a var and throw it after container.run(). Something like:
LocalConatinerRunner {
  Throwable throwable;
  ...
  container setContainerListener(new SamzaContainerListener() {
   ...
   onContainerFailed(Throwable t) {
     throwable = t;
  }
 }

container.run();

if(throwable != null) {
  throw throwable
}
...